src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1593,13 +1595,58 @@ impl_types layout_optimizer::get_preferred_impl_type(program_node& node, format
         if (node.get_output_layout().data_type == data_types::i32)
             return impl_types::ocl;
 
+        for (auto& dep : node.get_dependencies()) {
+            for (auto& user : dep.first->get_users()) {
+                if (user->is_type<convolution>() && (dep.first->id() == user->get_dependency(0).id())) {
+                    auto& conv_node = user->as<convolution>();
+                    auto needed_padding = prepare_padding::get_convolution_required_padding(conv_node);
+                    if (needed_padding) {
+                        return impl_types::ocl;
+                    }
+                }
+            }
+        }
+
+        for (auto usr : node.get_users()) {
+            if (usr->is_type<convolution>()) {
+                auto& conv_node = usr->as<convolution>();
+                auto needed_padding = prepare_padding::get_convolution_required_padding(conv_node);
+                if (needed_padding) {
+                    return impl_types::ocl;
+                }
+            }
+        }
+
         for (auto& dep : node.get_dependencies()) {
             if (dep.first->is_in_data_flow() && dep.first->get_preferred_impl_type() == impl_types::onednn) {
                 return impl_types::onednn;
             }
         }
+
         if (format::is_blocked(node.get_output_layout().format)) {
-            return impl_types::onednn;
+            auto concat_params = node.get_kernel_impl_params();
+            auto concat_axis = concat_params->typed_desc<concatenation>()->axis;
+            // Avoid onednn impl_type in shallow feature size.
```

Comment:
`// Avoid onednn impl_type in shallow feature size because of performance issue`